tests/main/snap-snap-uri-handler/task.yaml

```diff
@@ -0,0 +1,42 @@
+summary: Ensure the "snap snap-uri-handler" command works
+
+# We don't run on ubuntu-core, since we are testing the behaviour of
+# xdg-open on the host system.
+systems: [-ubuntu-core-*]
+
+# xdg-open doesn't seem to check desktop files if it doesn't think it
+# has a display.
+environment:
+    DISPLAY: :placeholder
+
+prepare: |
+    #shellcheck source=tests/lib/pkgdb.sh
+    . "$TESTSLIB/pkgdb.sh"
+    distro_install_package xdg-utils
+    touch /usr/bin/zenity
+
+restore: |
+    umount -f /usr/bin/zenity || :
+
+execute: |
+    echo "URI Handler fails if gnome-software is not installed and user refuses to install it"
+    mount --bind /bin/false /usr/bin/zenity
+    if snap snap-uri-handler snap://package 2>errors.log; then
+        cat errors.log >&2
+        echo "Expected URI handler to fail"
+        exit 1
+    fi
+    MATCH "GNOME Software required" < errors.log
+
+    echo "Now with gnome-software installed"
+    snap try "$TESTSLIB"/snaps/gnome-software
+    snap snap-uri-handler snap://package | MATCH "Open URI: snap://package"
```

Comment:
Not sure if this is acknowledged with regards to design but I find it repetitive. `snap uri-handler snap://package` would look nicer IMO.

Reply:
I guess I wanted to namespace the command to make it clear that it was for handling "snap://" URIs.

I don't think users would tend to call this directly, so ideally it'd be a hidden command.  Some of the "snap debug" commands were hidden, but I couldn't see how to do the same for top level commands.